PhysicsTools/Heppy/python/analyzers/objects/PhotonAnalyzer.py

```diff
@@ -135,7 +152,48 @@ def idWP(gamma,X):
         self.counters.counter('events').inc('all events')
         if foundPhoton: self.counters.counter('events').inc('has >=1 gamma at preselection')
         if len(event.selectedPhotons): self.counters.counter('events').inc('has >=1 selected gamma')
-       
+
+    def checkGenPhoton(self, event):
+        def isHardParton(particle):
+            return abs(particle.pdgId()) in [1,2,3,4,5,6,21] and particle.status() == 23
+
+        def isPromptPhoton(photon):
+            return photon.status() == 1 and abs(photon.mother(0).pdgId()) <= 22 and abs(photon.mother(0).pdgId()) not in [11,13,15]
```

Comment:
(Technically ISR and FSR photons can even interfere, so checking the mother of the photon in this way is not well defined). 

If you really must distinguish between "photons radiated from final state leptons" and other prompt photons, the more robust way would actually be to check delta R to the nearest status 1 lepton.  What is the motivation to call these photons non-prompt?